analytic_estimated_amount/models/purchase_order.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# © 2017 Savoir-faire Linux
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, models
+
+
+class PurchaseOrder(models.Model):
+
+    _inherit = 'purchase.order'
+
+    @api.multi
+    def unlink(self):
+        self._remove_analytic_lines()
+        return super(PurchaseOrder, self).unlink()
+
+    @api.multi
+    def button_confirm(self):
+        res = super(PurchaseOrder, self).button_confirm()
+        self.mapped('order_line')._generate_analytic_line()
+        return res
+
+    @api.multi
+    def button_cancel(self):
+        res = super(PurchaseOrder, self).button_cancel()
+        self._remove_analytic_lines()
+        return res
+
+    @api.multi
+    def _remove_analytic_lines(self):
+        analytic_lines = self.mapped('order_line.analytic_line_id')
+        if analytic_lines:
+            analytic_lines.unlink()
```

Comment:
Not a big deal, but is that necessary, given that _remove_analytic_lines is called on the lines when they are unlinked? so they will be deleted anyway if delete=cascade? (and if delete != cascade, does it make sense to keep the PO lines, but delete the analytic lines, when the PO is deleted?

EDIT: I see now it's called by button_cancel, not just unlink, having the method makes sense, so change my comment to: does it make sense to call it in unlink()?